### PR TITLE
ROS Parameters: Add boat simulator ROS parameters

### PR DESCRIPTION
**Pull request by chrischang5**
### Description

- Resolves https://github.com/UBCSailbot/boat_simulator/issues/37
- Implementation of reading these parameters in the `boat_simulator` is done in https://github.com/UBCSailbot/boat_simulator/pull/43

### Verification

- [ ] See https://github.com/UBCSailbot/boat_simulator/pull/43 for verification steps of reading in parameters in the `boat_simulator`

### Resources

- [ROS2 YAML parameter structure](https://roboticsbackend.com/ros2-yaml-params/#Why_using_YAML_for_ROS2_params)
- [ROS2 Parameter docs](https://docs.ros.org/en/humble/Concepts/Basic/About-Parameters.html)

### CODE REVIEW

**Review by patrick-5546 — APPROVED** [review]
LGTM infrastructure wise, I'll trust @DFriend01 to verify that it works

File: src/global_launch/config/globals.yaml
Comment (DFriend01) [inline src/global_launch/config/globals.yaml:1]: When using interval notation, let's try to be more mathematically specific (i.e. use `(` for non-inclusive boundaries and `[` for inclusive boundaries). Remember for infinity, use `(` (i.e. for `DOUBLE_MAX`, `DOUBLE_MIN`)

File: src/global_launch/config/globals.yaml [hunk 1]
```diff
@@ -2,3 +2,37 @@
...
       # Publishers' period (seconds): how often the publishers publish
       pub_period_sec: 0.5
+
+# boat_simulator parameters
+low_level_control_node:
+   ros__parameters:
+      logging_throttle_period_sec: 2.0 # [0.0, MAX_DOUBLE]
+      rudder:
+         disable_actuation: false
+         fixed_angle_deg: 0.0 # Only used if rudder actuation is disabled: [0.0, 360.0]
```
Comment (DFriend01) [inline src/global_launch/config/globals.yaml:12]: Range should be [-45, 45]

File: src/global_launch/config/globals.yaml [hunk 2]
```diff
@@ -2,3 +2,37 @@
...
+         actuation_execution_period_sec: 0.5 # How often the rudder action server routine's main loop executes: [0.0, MAX_DOubLE]
```
Comment (DFriend01) [inline src/global_launch/config/globals.yaml:13]: Should not include zero

File: src/global_launch/config/globals.yaml [hunk 3]
```diff
@@ -2,3 +2,37 @@
...
+         pid: # Only used if rudder actuation is enabled
+            kp: 0.0 # [0.0, MAX_DOUBLE]
+            ki: 0.0 # [0.0, MAX_DOUBLE]
+            kd: 0.0 # [0.0, MAX_DOUBLE]
+            buffer_size: 50 # [0, MAX_INT]
```
Comment (DFriend01) [inline src/global_launch/config/globals.yaml:18]: Should not include zero (start from 1)

File: src/global_launch/config/globals.yaml [hunk 4]
```diff
@@ -2,3 +2,37 @@
...
+      wingsail:
+         disable_actuation: false
+         fixed_angle_deg: 0.0 # Only used if actuation is disabled: [0.0, 360.0]
```
Comment (DFriend01) [inline src/global_launch/config/globals.yaml:21]: Let's make the range symmetric => [-180, 180)

File: src/global_launch/config/globals.yaml [hunk 5]
```diff
@@ -2,3 +2,37 @@
...
+         actuation_execution_period_sec: 0.5 # How often the sail action server routine's main loop executes: [0.0, MAX_DOUBLE]
+         actuation_speed_deg_per_sec: 0.1 # Only used if actuation is enable: [0.0, MAX_DOUBLE]
```
Comment (DFriend01) [inline src/global_launch/config/globals.yaml:23]: Should not include zero

File: src/global_launch/config/globals.yaml [hunk 6]
```diff
@@ -2,3 +2,37 @@
...
+physics_engine_node:
+   ros__parameters:
+      rudder:
+         actuation_request_period_sec: 10.0 # How often the rudder action client requests a rudder actuation: [0.0, MAX_DOUBLE]
+      wingsail:
+         actuation_request_period_sec: 10.0 # How often the sail action server requests a wingsail actuation: [0.0, MAX_DOUBLE]
```
Comment (DFriend01) [inline src/global_launch/config/globals.yaml:29]: Should not include zero

**Review by DFriend01 — CHANGES_REQUESTED** [review]
Just some comments on the parameter ranges.

File: src/global_launch/config/globals.yaml (hunk 3 above)
Comment (chrischang5) [reply src/global_launch/config/globals.yaml:18]: Will update value to reflect this then.

**Comment by DFriend01** [comment]
@chrischang5 I realize you are working on this right now, but can you create a `README.md` file inside of the `config` directory instead, of defining the configuration parameters?

For each parameter, can you add:
- An explanation of what the parameter is
- Its type
- The range of acceptable values

The config file is starting to look a little messy, so perhaps this would be a better approach to documenting these values. What do you think?

**Comment by chrischang5** [comment]
> @chrischang5 I realize you are working on this right now, but can you create a `README.md` file inside of the `config` directory instead of the configuration file itself?
> 
> For each parameter, can you add:
> 
> * An explanation of what the parameter is
> * Its type
> * The range of acceptable values
> 
> The config file is starting to look a little messy, so perhaps this would be a better approach to documenting these values. What do you think? I trust that you can devise an acceptable format for the README.

README seems like a better solution. One drawback is that it's a separate place we need to maintain alongside the config file, but I think its' better than a messy config file.

**Comment by chrischang5** [comment]
We should spend more time in the design phase next time to avoid changing implementations all the time. Might have made this process smoother.

**Comment by DFriend01** [comment]
I updated the configuration parameter descriptions. @patrick-5546 did you want to take a quick look?

**Comment by patrick-5546** [comment]
Looks great! A couple of suggestions
- Add the README to the docs site
- Alphabetize the parameters

**Review by DFriend01 — APPROVED** [review]

**Pull request merged**